### PR TITLE
[C-6187] Add support for token management API

### PR DESCRIPTION
**Pull request by drew-y**
## Description of the change

See title

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/CourierClient.php [hunk 1]
```diff
@@ -1121,4 +1121,77 @@ public function getAudiences(): object
...
+     */
+    public function getUserToken(string $user_id, string $token): object
+    {
+        return $this->doRequest(
+            $this->buildRequest("get", "users/" . $user_id . "/tokens/" . $token)
+        );
+    }
+
+    /**
+     * Apply a JSON Patch (RFC 6902) to the specified token.
```
Comment (tk26) [inline src/CourierClient.php]: copy pasta?

Comment (tk26) [inline src/CourierClient.php]:
```suggestion
     * Get the tokens associated with a user
```

File: tests/NotificationTest.php [hunk 2]
```diff
@@ -512,4 +513,48 @@ public function test_get_audiences() {
...
+
+    public function test_get_user_token() {
+        $response = $this->getCourierClient()->getUserToken("me", "my_token");
+
+        $this->assertEquals("GET", $response->method);
+        $this->assertEquals("application/json", $response->content);
+        $this->assertEquals("/users/me/tokens/my_token", $response->path);
+    }
+
+    public function test_patch_user_tokens() {
```
Comment (tk26) [inline tests/NotificationTest.php]: copy pasta? might also want to remove $patch?

Comment (tk26) [inline tests/NotificationTest.php]:
```suggestion
    public function test_get_user_tokens() {
```

File: tests/NotificationTest.php [hunk 3]
```diff
@@ -512,4 +513,48 @@ public function test_get_audiences() {
...
+        $patch = array(array("token" => "my_token"));
```
Comment (tk26) [inline tests/NotificationTest.php]:
```suggestion
```

File: tests/NotificationTest.php (hunk 2 above)
Comment (drew-y) [reply tests/NotificationTest.php]: Fixed. Thanks

**Review by tk26 — APPROVED** [review]

**Pull request merged**